doc/faq.md

```diff
@@ -57,6 +57,65 @@ directory. None of this should affect any existing Haskell tools at all.
   there may be more options here for new projects in the future (see issue
   [253](https://github.com/commercialhaskell/stack/issues/253))
 
+### Cabal-the-library
+
+The relationship between stack and Cabal-the-library is complicated.
+
+1. Stack itself builds against a version of the Cabal library, which it uses for parsing Cabal files.
+
+2. Stack wraps functionality from a version of the Cabal library in order to build packages.
+
+    i. For packages with `build-type: Simple`, Stack uses a setup executable compiled with a version of the Cabal library to perform builds. All packages using the same compiler and Cabal version are built with the same executable. These executables are cached in the `setup-exe-cache` configuration directory.
+
+    ii. For packages with `build-type: Custom`, Stack compiles `Setup.hs` against a version of the Cabal library and uses that setup executable to perform builds. The executable is stored TODO
```

Comment:
"against a version of the Cabal library"- did you mean to include "specified by your stack.yaml file"?

Reply:
I think this is covered later — so it probably makes more sense to zip these two lists of numbered paragraphs (the relationship and the versions) and have each bit of functionality in a self-contained unit.

Reply:
Fixed at https://github.com/dbaynard/stack/blob/f31e9dd962aefa6c11854496ff578244618a9623/doc/faq.md#L85-L87